src/ch/blinkenlights/android/vanilla/LibraryActivity.java

```diff
@@ -360,25 +361,35 @@ private void updateHeaders()
 	}
 
 	/**
-	 * Adds songs matching the data from the given intent to the song timelime.
+	 * Adds songs matching the data from the given intent to the song timeline.
 	 *
 	 * @param intent An intent created with
 	 * {@link LibraryAdapter#createData(View)}.
 	 * @param action One of LibraryActivity.ACTION_*
 	 */
-	private void pickSongs(Intent intent, int action)
+	private void pickSongs(Intent intent, final int action)
 	{
+		int effectiveAction = action; // mutable copy
 		long id = intent.getLongExtra("id", LibraryAdapter.INVALID_ID);
+		int type = mCurrentAdapter.getMediaType();
+
+		// special handling if we pick one song to be played that is already in queue
+		boolean songPicked = (id != LibraryAdapter.INVALID_ID && type == MediaUtils.TYPE_SONG);
```

Comment:
I just wonder when `id` would equal `LibraryAdapter.INVALID_ID`: This would only be the case if the intent didn't have any data for `"id"` which shouldn't be the case.

(At least, looking at the code i don't see that pickSongs() would get called with an intent which lacks the `"id"` extra)